### DISCUSSION TITLE
conflict in i2c drivers 71 and 72

### DISCUSSION THREAD

**Discussion by fb-pilot** [arendst/Tasmota#20579] [Q&A]
when activating the two i2c drivers together (71 + 72) sensor100 does not work --> RESULT = {"Command":"Error"}

### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): _wmos d1_
- [x] Tasmota binary firmware version number used:  [13.3.0.4]
  - [ ] Pre-compiled
  - [x] Self-compiled
- [x] Flashing tools used:   VSC as editor with Platformio
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
15:30:41.027 CMD: status 0
15:30:41.035 RSL: STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota",""],"Topic":"tasmota_5F5846","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
15:30:41.040 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota-4M.bin.gz","RestartReason":"Software/System restart","Uptime":"0T03:12:21","StartupUTC":"2024-01-24T11:18:20","Sleep":50,"CfgHolder":4617,"BootCount":15,"BCResetTime":"2024-01-24T11:28:28","SaveCount":29,"SaveAddress":"3FB000"}}
15:30:41.044 RSL: STATUS2 = {"StatusFWR":{"Version":"13.3.0.4(tasmota-4M)","BuildDateTime":"2024-01-24T12:13:37","Boot":31,"Core":"2_7_6","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"365/699"}}
15:30:41.048 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["bf-home-wlan1",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008001","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
15:30:41.060 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":722,"Free":1324,"Heap":22,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"1640EF","FlashFrequency":40,"FlashMode":"DOUT","Features":["0809","8F9AC78F","04368001","000000CF","010013C0","C000F981","00004084","00201000","54000020","00000088","00000000"],"Drivers":"1,!2,!3,4,!5,!6,!7,!8,9,10,!12,!16,!18,!19,!20,!21,!22,!24,26,!27,29,!30,!35,!37,38,!45,50,62,!68","Sensors":"1,2,3,4,5,6,33,100","I2CDriver":"7,71,72"}}
15:30:41.065 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-5F5846-6214","IPAddress":"192.168.192.25","Gateway":"192.168.192.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.192.1","DNSServer2":"0.0.0.0","Mac":"50:02:91:5F:58:46","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
15:30:41.070 RSL: STATUS7 = {"StatusTIM":{"UTC":"2024-01-24T14:30:41Z","Local":"2024-01-24T15:30:41","StartDST":"2024-03-31T02:00:00","EndDST":"2024-10-27T03:00:00","Timezone":"+01:00","Sunrise":"08:30","Sunset":"17:33"}}
15:30:41.117 RSL: STATUS10 = {"StatusSNS":{"Time":"2024-01-24T15:30:41","Switch1":"OFF","COUNTER":{"C1":0},"ANALOG":{"A0":6}}}
15:30:41.124 RSL: STATUS11 = {"StatusSTS":{"Time":"2024-01-24T15:30:41","Uptime":"0T03:12:21","UptimeSec":11541,"Heap":22,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":27,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Dimmer":10,"Color":"1A1A1A","HSBColor":"0,0,10","Channel":[10,10,10],"Scheme":0,"Width":1,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"bf-home-wlan1","BSSId":"E0:28:6D:96:51:70","Channel":6,"Mode":"11n","RSSI":100,"Signal":-50,"LinkCount":1,"Downtime":"0T00:00:04"}}}

```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
00:00:00.001 HDW: ESP8266EX
00:00:00.007 UFS: FlashFS mounted with 1984 kB free
00:00:00.055 CFG: Loaded from File, Count 26
00:00:00.060 FRC: Some settings have been reset (2)
00:00:00.077 Project tasmota - Tasmota Version 13.3.0.4(tasmota-4M)-2_7_6(2024-01-24T12:13:37)
00:00:00.082 I2C: INA3221 found at 0x40
00:00:01.001 WIF: Connecting to AP1 bf-home-wlan1 Channel 6 BSSId 
E0:28:6D:96:51:70 in mode 11n as tasmota-5F5846-6214...
00:00:03.566 WIF: Connected
00:00:03.820 HTP: Web server active on tasmota-5F5846-6214 with IP address 192.168.192.25
00:00:03.823 RSL: INFO1 = {"Info1":{"Module":"Generic","Version":"13.3.0.4(tasmota-4M)","FallbackTopic":"cmnd/DVES_5F5846_fb/","GroupTopic":"cmnd/tasmotas/"}}
00:00:03.833 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-5F5846-6214","IPAddress":"192.168.192.25"}}
00:00:03.843 RSL: INFO3 = {"Info3":{"RestartReason":"External System","BootCount":14}}
00:00:03.851 RSL: RESULT = {"POWER1":"OFF"}
00:00:03.854 RSL: POWER1 = OFF
00:00:03.857 RSL: RESULT = {"POWER2":"OFF"}
00:00:03.861 RSL: POWER2 = OFF
12:15:18.267 QPC: Reset
12:15:20.244 RSL: STATE = {"Time":"2024-01-24T12:15:20","Uptime":"0T00:00:09","UptimeSec":9,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":33,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Dimmer":10,"Color":"1A1A1A","HSBColor":"0,0,10","Channel":[10,10,10],"Scheme":0,"Width":1,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"bf-home-wlan1","BSSId":"E0:28:6D:96:51:70","Channel":6,"Mode":"11n","RSSI":100,"Signal":-48,"LinkCount":1,"Downtime":"0T00:00:05"}}
12:15:20.317 RSL: SENSOR = {"Time":"2024-01-24T12:15:20","Switch1":"OFF","COUNTER":{"C1":0},"ANALOG":{"A0":3}}
i2cdriver
12:17:20.774 CMD: i2cdriver     
12:17:20.779 RSL: RESULT = {"I2CDriver":"7,!71,72"}
sensor100 1,0.1,0.1,0.1
12:17:42.886 CMD: sensor100 1,0.1,0.1,0.1
12:17:42.892 RSL: RESULT = {"Sensor100":{"idx":1,"addr":"0x40","rshunt":[0.10000,0.10000,0.10000]}}
␖
12:17:58.393 CMD: 
i2cdriver
12:18:06.517 CMD: i2cdriver
12:18:06.522 RSL: RESULT = {"I2CDriver":"7,!71,72"}
i2cdriver71 1
12:18:18.362 CMD: i2cdriver71 1
12:18:18.367 RSL: RESULT = {"I2CDriver":"7,71,72"}
12:18:19.566 APP: Restarting

 ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x4010f000, len 3584, room 16 
tail 0
chksum 0xb0
csum 0xb0
v3969889e
~ld

00:00:00.002 HDW: ESP8266EX
00:00:00.009 UFS: FlashFS mounted with 1984 kB free
00:00:00.057 CFG: Loaded from File, Count 28
00:00:00.062 QPC: Count 1
00:00:00.080 Project tasmota - Tasmota Version 13.3.0.4(tasmota-4M)-2_7_6(2024-01-24T12:13:37)
00:00:00.085 I2C: INA3221 found at 0x40
00:00:01.001 WIF: Connecting to AP1 bf-home-wlan1 Channel 6 BSSId E0:28:6D:96:51:70 in mode 11n as tasmota-5F5846-6214...
00:00:02.569 WIF: Connected
00:00:02.822 HTP: Web server active on tasmota-5F5846-6214 with IP address 192.168.192.25
00:00:02.825 RSL: INFO1 = {"Info1":{"Module":"Generic","Version":"13.3.0.4(tasmota-4M)","FallbackTopic":"cmnd/DVES_5F5846_fb/","GroupTopic":"cmnd/tasmotas/"}}
00:00:02.835 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-5F5846-6214","IPAddress":"192.168.192.25"}}
00:00:02.845 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":15}}
00:00:02.853 RSL: RESULT = {"POWER1":"OFF"}
00:00:02.857 RSL: POWER1 = OFF
00:00:02.860 RSL: RESULT = {"POWER2":"OFF"}
00:00:02.864 RSL: POWER2 = OFF
12:18:27.255 QPC: Reset
12:18:28.239 RSL: STATE = {"Time":"2024-01-24T12:18:28","Uptime":"0T00:00:08","UptimeSec":8,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":37,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Dimmer":10,"Color":"1A1A1A","HSBColor":"0,0,10","Channel":[10,10,10],"Scheme":0,"Width":1,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"bf-home-wlan1","BSSId":"E0:28:6D:96:51:70","Channel":6,"Mode":"11n","RSSI":96,"Signal":-52,"LinkCount":1,"Downtime":"0T00:00:04"}}
12:18:28.313 RSL: SENSOR = {"Time":"2024-01-24T12:18:28","Switch1":"OFF","COUNTER":{"C1":0},"ANALOG":{"A0":3}}
i2cdriver
12:18:41.603 CMD: i2cdriver
12:18:41.608 RSL: RESULT = {"I2CDriver":"7,71,72"}
i2cdriver
12:18:51.619 CMD: i2cdriver
12:18:51.624 RSL: RESULT = {"I2CDriver":"7,71,72"}
␖
12:18:58.755 CMD: 
sensor100 1,0.1,0.1,0.1
12:19:01.728 CMD: sensor100 1,0.1,0.1,0.1
12:19:01.732 RSL: RESULT = {"Command":"Error"}

```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
Disable the not needed driver https://tasmota.github.io/docs/I2CDEVICES/
When i2c hardware uses the same addresses it conflicts. So not every driver combination is possible
This is not a bug. Limitation of hardware.

**Reply by sfromis** [reply]
Also note that you may have a serious issue in your setup when you get this message. When detecting a boot loop, Tasmota will start to disable features until able to boot without quickly crashing again.
> `00:00:00.060 FRC: Some settings have been reset (2)`

Better get this resolved before next steps in trying to make it work.

However, it does not look like I2C address conflict, with INA3221 on 0x40-0x43 and QMC5883L on 0x0D.

To be clear, the latter does not mean "no conflict", as you should see a message about QMC5883L being detected if everything was working. It can be useful to run the `I2CScan` command to verify what devices are seen on the bus.

**Reply by fb-pilot** [reply]
thank you for your quick answer and your excelent work !
There is only the ina3221 on the bus
17:23:06.158 CMD: i2cscan
17:23:06.177 RSL: RESULT = {"I2CScan":"Device(s) found at 0x40"}
so I think there should not be a i2c bus conflict
the warnig "Some settings have been reset (2)" only apears at the first boot after modifying and loading the firmware - the second boot after enabling the driver 71 provides no warning.
I just added the two i2cdriver to the "tasmota\user_config_override.h" by adding "#define USE_QMC5883L
#define USE_INA3221" and translated the environment "tasmota-4M" lpading to the wemos-d1 module with a lonly ina3221 conected.
After booting the both drivers ar activated by default - so the Web shows only the ina3221 header and the command "sensor100" returns error.
After disabling the driver 71 the "sensor100" command works as expected and the web shows all readings of the ina 3221 as expected.
What else can I test to locate this issue ?

**Reply by barbudor** [reply]
Can you please try the following:
- Set WebLog 3
- Disable every I2C drivers except Driver 72 (INA3221) => Tasmota will restart
- Wait for the 1st tele/STATE & tele/SENSOR message
- Test if `Sensor100`  works
- Provide the console logs upon restart from `HDW: ESP8266EX` up to the response to the `Sensor100` command

Then
- Enabled I2Cdriver 71 (QMC) => Tasmota will restart
- Wait for the 1st tele/STATE & tele/SENSOR message
- Test if `Sensor100`  works
- Provide the console logs upon restart from `HDW: ESP8266EX` up to the response to the `Sensor100` command

Thanks

**Reply by fb-pilot** [reply]
ihats what I did - the other way around - except log level3
here is the outpu part 1
00:00:00.001 HDW: ESP8266EX
00:00:00.012 UFS: FlashFS mounted with 1984 kB free
00:00:00.062 CFG: Loaded from File, Count 33
00:00:00.068 QPC: Count 1
00:00:00.070 CFG: CR 365/699, Busy 0
00:00:00.075 CFG: CR 365/699, Busy 0
00:00:00.081 ROT: Mode 1
00:00:00.083 SRC: Restart
00:00:00.086 Project tasmota - Tasmota Version 13.3.0.4(tasmota-4M)-2_7_6(2024-01-24T12:13:37)
00:00:00.088 DSB: Sensors found 0
00:00:00.091 I2C: INA3221 found at 0x40
00:00:00.193 WIF: Attempting connection...
00:00:01.001 WIF: Connecting to AP1 bf-home-wlan1 Channel 6 BSSId E0:28:6D:96:51:70 in mode 11n as tasmota-5F5846-6214...
00:00:02.568 WIF: Attempting connection...
00:00:03.571 WIF: Connected
00:00:03.823 HTP: Web server active on tasmota-5F5846-6214 with IP address 192.168.192.25
00:00:03.827 RSL: INFO1 = {"Info1":{"Module":"Generic","Version":"13.3.0.4(tasmota-4M)","FallbackTopic":"cmnd/DVES_5F5846_fb/","GroupTopic":"cmnd/tasmotas/"}}
00:00:03.836 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-5F5846-6214","IPAddress":"192.168.192.25"}}
00:00:03.847 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":17}}
00:00:03.855 RSL: RESULT = {"POWER1":"OFF"}
00:00:03.858 RSL: POWER1 = OFF
00:00:03.861 RSL: RESULT = {"POWER2":"OFF"}
00:00:03.865 RSL: POWER2 = OFF
00:00:03.872 TFS: File 'autoexec.bat' not found
00:00:04.252 RTC: UTC 2024-01-24T18:15:03Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
19:15:03.000 RTC: Synced by NTP
19:15:05.248 QPC: Reset
19:15:07.239 APP: Boot Count 17
19:15:07.245 RSL: STATE = {"Time":"2024-01-24T19:15:07","Uptime":"0T00:00:10","UptimeSec":10,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":24,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Dimmer":10,"Color":"1A1A1A","HSBColor":"0,0,10","Channel":[10,10,10],"Scheme":0,"Width":1,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"bf-home-wlan1","BSSId":"E0:28:6D:96:51:70","Channel":6,"Mode":"11n","RSSI":98,"Signal":-51,"LinkCount":1,"Downtime":"0T00:00:05"}}
19:15:07.318 RSL: SENSOR = {"Time":"2024-01-24T19:15:07","Switch1":"OFF","COUNTER":{"C1":0},"ANALOG":{"A0":3}}
19:15:07.500 CFG: Saved to flash at 3FB, Count 34, Bytes 4096
19:16:17.151 CMD: i2cdriver
19:16:17.153 SRC: WebConsole from 192.168.192.20
19:16:17.155 CMD: Grp 0, Cmd 'I2CDRIVER', Idx 1, Len 0, Pld -99, Data ''
19:16:17.158 RSL: RESULT = {"I2CDriver":"!7,!71,72"}
19:16:29.237 CMD: sensor100 1,0.1,0.1,0.1
19:16:29.239 SRC: WebConsole from 192.168.192.20
19:16:29.241 CMD: Grp 0, Cmd 'SENSOR', Idx 100, Len 13, Pld 1, Data '1,0.1,0.1,0.1'
19:16:29.245 RSL: RESULT = {"Sensor100":{"idx":1,"addr":"0x40","rshunt":[0.10000,0.10000,0.10000]}}

part 2
00:00:00.001 HDW: ESP8266EX
00:00:00.013 UFS: FlashFS mounted with 1984 kB free
00:00:00.063 CFG: Loaded from File, Count 35
00:00:00.069 QPC: Count 1
00:00:00.071 CFG: CR 365/699, Busy 0
00:00:00.076 CFG: CR 365/699, Busy 0
00:00:00.082 ROT: Mode 1
00:00:00.084 SRC: Restart
00:00:00.087 Project tasmota - Tasmota Version 13.3.0.4(tasmota-4M)-2_7_6(2024-01-24T12:13:37)
00:00:00.089 DSB: Sensors found 0
00:00:00.092 I2C: INA3221 found at 0x40
00:00:00.195 WIF: Attempting connection...
00:00:01.001 WIF: Connecting to AP1 bf-home-wlan1 Channel 6 BSSId E0:28:6D:96:51:70 in mode 11n as tasmota-5F5846-6214...
00:00:02.568 WIF: Attempting connection...
00:00:03.573 WIF: Connected
00:00:03.826 HTP: Web server active on tasmota-5F5846-6214 with IP address 192.168.192.25
00:00:03.830 RSL: INFO1 = {"Info1":{"Module":"Generic","Version":"13.3.0.4(tasmota-4M)","FallbackTopic":"cmnd/DVES_5F5846_fb/","GroupTopic":"cmnd/tasmotas/"}}
00:00:03.839 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-5F5846-6214","IPAddress":"192.168.192.25"}}
00:00:03.849 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":18}}
00:00:03.858 RSL: RESULT = {"POWER1":"OFF"}
00:00:03.861 RSL: POWER1 = OFF
00:00:03.864 RSL: RESULT = {"POWER2":"OFF"}
00:00:03.868 RSL: POWER2 = OFF
00:00:03.877 TFS: File 'autoexec.bat' not found
00:00:05.651 NTP: No reply from 78.47.118.0
00:00:05.729 RTC: UTC 2024-01-24T18:20:30Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
19:20:30.000 RTC: Synced by NTP
19:20:30.253 QPC: Reset
19:20:32.211 APP: Boot Count 18
19:20:32.217 RSL: STATE = {"Time":"2024-01-24T19:20:32","Uptime":"0T00:00:09","UptimeSec":9,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":39,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Dimmer":10,"Color":"1A1A1A","HSBColor":"0,0,10","Channel":[10,10,10],"Scheme":0,"Width":1,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"bf-home-wlan1","BSSId":"E0:28:6D:96:51:70","Channel":6,"Mode":"11n","RSSI":98,"Signal":-51,"LinkCount":1,"Downtime":"0T00:00:05"}}
19:20:32.291 RSL: SENSOR = {"Time":"2024-01-24T19:20:32","Switch1":"OFF","COUNTER":{"C1":0},"ANALOG":{"A0":3}}
19:20:32.878 CFG: Saved to flash at 3FB, Count 36, Bytes 4096
19:21:35.809 CMD: i2cdriver
19:21:35.811 SRC: WebConsole from 192.168.192.20
19:21:35.813 CMD: Grp 0, Cmd 'I2CDRIVER', Idx 1, Len 0, Pld -99, Data ''
19:21:35.817 RSL: RESULT = {"I2CDriver":"!7,71,72"}
19:21:39.818 CMD: sensor100 1,0.1,0.1,0.1
19:21:39.820 SRC: WebConsole from 192.168.192.20
19:21:39.822 CMD: Grp 0, Cmd 'SENSOR', Idx 100, Len 13, Pld 1, Data '1,0.1,0.1,0.1'
19:21:39.825 RSL: RESULT = {"Command":"Error"}

hope it helps - good luck

**Reply by barbudor** [reply]
Thanks
Very strange
I will investigate this tonight or tomorrow evening

**Reply by fb-pilot** [reply]
dabke :-))

**Reply by barbudor** [reply]
@fb-pilot 
Problem identified and solved in the QMC5883 driver (as well as the HMC5883 driver).
PR #20585  done
Wait for the merge then update your code to the latest dev

**Reply by fb-pilot** [reply]
I hope your night was'nt too bad ;-)
perfect work - issue solved !
Thanks a lot 👍

**Comment by Jason2866** [comment]
If you compile a version just with the driver for the QMC5883, does the driver for the sensor work?

**Reply by fb-pilot** [reply]
I wrote an answer yesterday - seems to disapeared !? :-(
I dont have the QMC5883 conected yet - so I can try in some days - if useful .

**Reply by sfromis** [reply]
Well, barbudor already debugged the issue and submitted a correction to the QMC5883 driver. By now, this has been merged, and you can update and build again.

**Reply by fb-pilot** [reply]
did it - and works perfect !
super job - Thanks 👍

**Reply by fb-pilot** [reply]
now I conected the QMC5883 - and works perfect !
Thanks to all 👍
